app/src/main/java/com/keylesspalace/tusky/fragment/ViewImageFragment.kt

```diff
@@ -108,95 +111,129 @@ class ViewImageFragment : ViewMediaFragment() {
             }
         }
 
-        attacher = PhotoViewAttacher(binding.photoView).apply {
-            // This prevents conflicts with ViewPager
-            setAllowParentInterceptOnEdge(true)
+        val singleTapDetector = GestureDetectorCompat(
+            requireContext(),
+            object : GestureDetector.SimpleOnGestureListener() {
+                override fun onDown(e: MotionEvent) = true
+                override fun onSingleTapConfirmed(e: MotionEvent): Boolean {
+                    photoActionsListener.onPhotoTap()
+                    return false
+                }
+            }
+        )
 
-            // Clicking outside the photo closes the viewer.
-            setOnOutsidePhotoTapListener { photoActionsListener.onDismiss() }
-            setOnClickListener { onMediaTap() }
+        binding.photoView.setOnTouchCoordinatesListener(object : OnTouchCoordinatesListener {
+            var lastY: Float? = null
+            override fun onTouchCoordinate(view: View, event: MotionEvent, bitmapPoint: PointF) {
+                singleTapDetector.onTouchEvent(event)
 
-            /* A vertical swipe motion also closes the viewer. This is especially useful when the photo
-             * mostly fills the screen so clicking outside is difficult. */
-            setOnSingleFlingListener { _, _, velocityX, velocityY ->
-                var result = false
-                if (abs(velocityY) > abs(velocityX)) {
-                    photoActionsListener.onDismiss()
-                    result = true
+                // Two fingers have gone down after a single finger drag. Finish the drag
+                if (event.pointerCount == 2 && lastY != null) {
+                    onGestureEnd(view)
+                    lastY = null
                 }
-                result
-            }
-        }
 
-        var lastY = 0f
+                // The user is starting or stopping a pinch-zoom. If starting then disable touch
+                // events on the parent, so it does not attempt to scroll horizontally. If stopping
+                // then re-enable, to allow the user to swipe horizontally.
+                if (event.pointerCount >= 2 || view.canScrollHorizontally(1) && view.canScrollHorizontally(-1)) {
```

Comment:
The two `canScrollHorizontally` are an "is zoomed and not edge"?

Reply:
Yeah. I've updated the comment to make this clearer.

Reply:
Would've introduced a variable `isZoomedAndNotEdge` :-)